diff --git a/compat/backport-3.13.c b/compat/backport-3.13.c
index 6ce72b2..aad5edb 100644
--- a/compat/backport-3.13.c
+++ b/compat/backport-3.13.c
@@ -258,6 +258,7 @@ bool pci_device_is_present(struct pci_dev *pdev)
 EXPORT_SYMBOL_GPL(pci_device_is_present);
 #endif /* CONFIG_PCI */
 
+#if 0
 #ifdef CONFIG_HWMON
 struct device*
 hwmon_device_register_with_groups(struct device *dev, const char *name,
@@ -307,3 +308,4 @@ error:
 }
 EXPORT_SYMBOL_GPL(devm_hwmon_device_register_with_groups);
 #endif
+#endif
